cli/src/main/java/org/jboss/sbomer/cli/feature/sbom/adjuster/SyftImageAdjuster.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.sbomer.cli.feature.sbom.adjuster;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.cyclonedx.model.Bom;
+import org.cyclonedx.model.Component;
+import org.cyclonedx.model.Dependency;
+import org.cyclonedx.model.Property;
+import org.jboss.sbomer.core.features.sbom.utils.SbomUtils;
+import org.jboss.sbomer.core.features.sbom.utils.UrlUtils;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import lombok.extern.slf4j.Slf4j;
+
+@ApplicationScoped
+@Slf4j
+public class SyftImageAdjuster implements Adjuster {
+    @Override
+    public Bom adjust(Bom bom) {
+        Component productComponent = bom.getMetadata().getComponent();
+
+        // Initialize properties for main component, if not done so yet
+        if (productComponent.getProperties() == null) {
+            productComponent.setProperties(new ArrayList<>());
+        }
+
+        // If there are properties in the metadata field of the manifest, move these into the main component's
+        // properties.
+        if (bom.getMetadata().getProperties() != null) {
+            log.debug(
+                    "Moving '{}' properties from metadata to main component",
+                    bom.getMetadata().getProperties().size());
+
+            System.out.println(bom.getMetadata().getProperties().toString());
+            productComponent.getProperties().addAll(bom.getMetadata().getProperties());
+            bom.getMetadata().setProperties(null);
+        }
+
+        // Cleanup main component
+        cleanupComponent(productComponent);
+
+        // ...and all other components
+        bom.getComponents().forEach(component -> {
+            cleanupComponent(component);
+        });
+
+        // Set the purl for the main component
+        productComponent.setPurl(
+                new StringBuilder("pkg:oci/").append(productComponent.getName().split("/")[2])
+                        .append("@")
+                        .append(UrlUtils.urlencode(productComponent.getVersion()))
+                        .toString());
+
+        // Remove all components that are not on the paths we are interested in
+        bom.getComponents().removeIf(c -> {
```

Comment:
My way of thinking is that I wanted to rename the props first and operate on `sbomer:` prefixed properties  instead of the un-processed `syft:` ones. But you are making good point here, so I'll adjust it.